src/main/java/appeng/items/materials/ItemMaterial.java

```diff
@@ -311,17 +312,14 @@ else if( te instanceof IUpgradeableHost )
 
 				if( u != null )
 				{
-					final InventoryAdaptor ad = InventoryAdaptor.getAdaptor( upgrades, EnumFacing.UP );
-					if( ad != null )
+					final InventoryAdaptor ad = new AdaptorItemHandler( upgrades );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the declaration of "ad" closer to the code that uses it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1941)